csrc/device_lower/pass/expr_sort.cpp

```diff
@@ -1506,10 +1506,24 @@ void ExprSegmentationSorter::sort() {
   // Need this for initialization of the DAG that is processed
   std::unordered_map<Expr*, ExprGroup*> expr2group;
 
-  // Not putting the exprs between allKnownVals() and fusion inputs here
+  // We skip generating code that computes only pointer-arithmetic outputs.
+  // Those outputs will be computed by ExpressionEvaluator.
+  std::vector<Val*> non_pointer_arithmetic_outs;
+  non_pointer_arithmetic_outs.reserve(fusion_->outputs().size());
+  std::copy_if(
+      fusion_->outputs().begin(),
+      fusion_->outputs().end(),
```

Comment:
QQ: should we use `getTerminatingOutputs()` here instead?

I guess that doesn't really make a difference to me.... i.e. `getExprsBetween` should be able to still find it.